ydb/docs/en/core/dev/example-app/python/index.md

```diff
@@ -21,217 +21,513 @@ Next, from the same working directory, run the command to start the test app. Th
+
+- Synchronous
+
+   ```python
+   def explicit_transaction_control(pool: ydb.QuerySessionPool, series_id, season_id, episode_id):
+       def callee(session: ydb.QuerySessionSync):
+           query = """
+           DECLARE $seriesId AS Int64;
+           DECLARE $seasonId AS Int64;
+           DECLARE $episodeId AS Int64;
+
+           UPDATE episodes
+           SET air_date = CurrentUtcDate()
+           WHERE series_id = $seriesId AND season_id = $seasonId AND episode_id = $episodeId;
+           """
+
+           # Get newly created transaction id
+           tx = session.transaction(ydb.QuerySerializableReadWrite()).begin()
+
+           # Execute data query.
+           # Transaction control settings continues active transaction (tx)
+           with tx.execute(
+               query,
+               {
+                   "$seriesId": (series_id, ydb.PrimitiveType.Int64),
+                   "$seasonId": (season_id, ydb.PrimitiveType.Int64),
+                   "$episodeId": (episode_id, ydb.PrimitiveType.Int64),
+               },
+           ) as _:
+               pass
+
+           print("\n> explicit TCL call")
+
+           # Commit active transaction(tx)
+           tx.commit()
+
+       return pool.retry_operation_sync(callee)
+   ```
+
+- Asynchronous
+
+   ```python
+   async def explicit_transaction_control(
+       pool: ydb.aio.QuerySessionPoolAsync, series_id, season_id, episode_id
+   ):
+       async def callee(session: ydb.aio.QuerySessionAsync):
+           query = """
+           DECLARE $seriesId AS Int64;
+           DECLARE $seasonId AS Int64;
+           DECLARE $episodeId AS Int64;
+
+           UPDATE episodes
+           SET air_date = CurrentUtcDate()
+           WHERE series_id = $seriesId AND season_id = $seasonId AND episode_id = $episodeId;
+           """
+
+           # Get newly created transaction id
+           tx = await session.transaction(ydb.QuerySerializableReadWrite()).begin()
+
+           # Execute data query.
+           # Transaction control settings continues active transaction (tx)
+           async with await tx.execute(
+               query,
+               {
+                   "$seriesId": (series_id, ydb.PrimitiveType.Int64),
+                   "$seasonId": (season_id, ydb.PrimitiveType.Int64),
+                   "$episodeId": (episode_id, ydb.PrimitiveType.Int64),
+               },
+           ) as _:
+               pass
+
+           print("\n> explicit TCL call")
+
+           # Commit active transaction(tx)
+           await tx.commit()
+
+       return await pool.retry_operation_async(callee)
+   ```
+
+{% endlist %}
+
+However, it is worth remembering that a transaction can be opened implicitly at the first request. It could be commited automatically with the explicit indication of the `commit_tx=True` flag.
+Implicit transaction management is preferable because it requires fewer server calls. An example of implicit control will be demonstrated in the next block.
```

Comment:
```suggestion
However, a transaction can be opened implicitly with the first request and can be committed automatically by setting the `commit_tx=True` flag in arguments. Implicit transaction management is preferable because it requires fewer server calls.
```

Let's add a separate set of examples for implicit transaction management. Demonstrating both implicit transaction management and result iteration in the same example below is confusing. For instance, it is unclear if there is another difference between `pool.execute_with_retries` and `tx.execute`: does the former commit by default, while the latter does not?